### ISSUE TITLE
sidebar: Separate folder name and theme names

### ISSUE THREAD

**Issue by nana-4** [themix-project/oomox#188]
This improves readability and avoids spreading the sidebar wide.

**Comment by actionless** [comment]
the second issue was already fixed in https://github.com/themix-project/oomox/issues/189

**Comment by actionless** [comment]
now i've attempted to address the first part of the issue, to improve the readability

please have a look on the latest `oomox-git`

**Comment by nana-4** [comment]
Thanks. I tried the latest oomox and it seems to have better readability. However, I still feel that it's better to separate the folder and themes completely, because:

- "shiki-brave" looks like a special (or meaningful) theme of "GNOME-Colors", for example.
- And it reduces the discoverability of the themes in expander.

Also, when separating the folder and themes, I suggest to unbold the (sub) folders. (I think it's enough to bold only the top folders. Otherwise, the sub-folders would look noisy.)

In addition (a bit off-topic, though),

- I think there is no need to make the text color dim in the sidebar anymore. The dim text is hard to read and looks like unclickable.
- The current minimum width of sidebar (`250`) looks a bit too wide. I think `200` is just right.
- I still think that classifying themes into "Presets" and "User Presets" without "Plugins" is easy for end users to understand. (IMHO, "Plugins" is not very clear for end users.)

In summary, it looks like:

```
v Presets
   > Base16
   v Contributed
      Christmas
      Cxmas
      Halloween
      QuakeWorld
      ...
   > GNOME-Colors
   > LCARS
   > Modern
   > Monovedek
   > Pop Art
   > Retro
   > WoopWoop
   > Xresources
   Random
v User Presets
   > Base16
   > Image Colors
   MyTheme
   MyTheme2
   MyTheme3
```

- Separate folder and themes completely
- Do not set sub-folders to bold
- Do not set text colors to dim
- Move "Random" into "Presets"
- Move "Image Colors" into "User Presets"
- Move "Base16" into "Presets" and user's "Base16" into "User Presets"

**Comment by actionless** [comment]
> Separate folder and themes completely

that reduces usability because it's not possible anymore to preview directory without opening, i mentioned that previously

>Do not set sub-folders to bold
>Do not set text colors to dim
>Set the minimum width of sidebar to 200

ok

> Move $PLUGIN into "Presets"

no, that would be totally illogical since there are different options for them than for normal presets

**Comment by nana-4** [comment]
> that reduces usability because it's not possible anymore to preview directory without opening, i mentioned that previously

Honestly, I still don't understand why you think it (markedly?) reduces usability. I think the two issues I mentioned above are more problematic. AFAIK and IIRC, other apps (not limited to GTK apps) don't have the pattern Oomox has. For example, in `gtk3-demo`'s sidebar, "Entry" is not "Entry: Entry Buffer". But do you think it is also reducing the usability?

As a side note, just FYI:

- In `gitg`, some directories are expanded but others are not expanded by default.
- In `atom` (non GTK app), directories are expanded by clicking the row.

Maybe can we use such approaches to make up for usability?

> no, that would be totally illogical since there are different options for them than for normal presets

Considering that "Presets" are themes prepared by Oomox and "User Presets" are themes created by the user, I thought that would be logical from the viewpoint of a user. But OK, I'll leave it up to your judgment in regards to that.

**Comment by actionless** [comment]
> For example, in gtk3-demo's sidebar, "Entry" is not "Entry: Entry Buffer". But do you think it is also reducing the usability?

i think for gtk3-demo such pattern (scrolling presets with keyboard arrows to preview the palettes) is not appliable since you need to click Run to actually see the demo

other examples also not answering my concern about limitations for keyboard navigation

> Considering that "Presets" are themes prepared by Oomox and "User Presets" are themes created by the user

yeah but plugins are neither of both -- contents of plugins are populated by users but can't be edited (the same as default presets) and stored in original (.yml, .png) format, not as themix-colors

**Comment by nana-4** [comment]
> i think for gtk3-demo such pattern (scrolling presets with keyboard arrows to preview the palettes) is not appliable since you need to click Run to actually see the demo

Hmm, I thought gtk3-demo's right pane is equivalent to Oomox's palettes (both can be previewed with arrow keys), and gtk-demo's "Run" button is equivalent to Oomox's "Export" buttons...

By the way, to expand directories, `<` and `>` arrow keys can be used.

> yeah but plugins are neither of both -- contents of plugins are populated by users but can't be edited (the same as default presets) and stored in original (.yml, .png) format, not as themix-colors

I see. Now I got it. Thanks for your detailed explanation :)

**Comment by actionless** [comment]
> gtk3-demo

i mean what information which reveals when item is selected in gtk3-demo is not so important for understanding the content of the item, while in oomox app just name of the preset tells not much and it need to be opened first to understand what's inside

> By the way, to expand directories, < and > arrow keys can be used.

please check the keyboard shortcuts popup, they should be already there
